### PR TITLE
Fix grouping of an empty frame

### LINKED ISSUE #2078: Internal error when groupping an empty frame
```
>>> import datatable as dt
>>> from datatable import f, by, count
>>> 
>>> DT = dt.Frame([[]])
>>> DT[:, count(f[0]), by(0)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Assertion 'max < rii->length || max == RowIndex::NA' failed in c/rowindex_array.cc, line 365
```

### PR DESCRIPTION
**Pull request by st-pasha**
Closes #2078

### CODE REVIEW

File: tests/test-groups.py [hunk 1]
```diff
@@ -30,89 +30,115 @@
...
-def test_groups_internal2():
-    d0 = dt.Frame([[1,   5,   3,   2,   1,    3,   1,   1,   None],
+def test_groups1():
+    DT = dt.Frame([[1,   5,   3,   2,   1,    3,   1,   1,   None],
                    ["a", "b", "c", "a", None, "f", "b", "h", "d"]],
                   names=["A", "B"])
-    d1 = d0[:, :, by("A")]
-    # gb = d1.internal.groupby
-    # assert gb.ngroups == 5
-    # assert gb.group_sizes == [1, 4, 1, 2, 1]
```
Comment (oleksiyskononenko) [inline tests/test-groups.py]: btw, how to get access to number of groups in python since `.internal` is gone?

Comment (st-pasha) [reply tests/test-groups.py]: well, that property no longer exists, so there is no way to access it.

Comment (oleksiyskononenko) [reply tests/test-groups.py]: Yeah, but sometimes it is nice to know number of groups.

Comment (st-pasha) [reply tests/test-groups.py]: It sure is! So maybe we can add some kind of helper function to access this and similar properties. Though I'm not sure how it's supposed to work API-wise

**Pull request merged**